airflow/api/client/json_client.py

```diff
@@ -24,9 +24,25 @@
 
 
 class Client(api_client.Client):
-    """Json API client implementation."""
+    """Json API client implementation.
 
-    def _request(self, url, method="GET", json=None):
+    This client is used to interact with a Json API server and perform various actions
+    such as triggering DAG runs,deleting DAGs, interacting with pools, and getting lineage information.
+
+    :param _api_base_url: The base URL for the Json API server.
+    :param _session: A session object to use for making HTTP requests.
+    """
+
+    def _request(self, url: str, json=None, method: str = "GET"):
+        """Make a request to the Json API server.
+
+        :param url: The URL to send the request to.
+        :param method: The HTTP method to use (e.g. "GET", "POST", "DELETE").
+        :param json: A dictionary containing JSON data to send in the request body.
+        :return: A dictionary containing the JSON response from the server.
+        :rtype: dict
```

Comment:
You can use typing instead of `:return:` and `:rtype:`, I think there's an `APIResponse` return type at `airflow/api_connexion/types`

Reply:
Absolutely agree with you, @ephraimbuddy, but in that case I receive following message from pre-commit run-mypy
hook. Should we leave this method this without typehint?
 
![Screenshot from 2022-12-28 18-49-43](https://user-images.githubusercontent.com/94890848/209837921-b2e560c0-8dc3-4fdc-bfa2-792551e199bc.png)

Reply:
Usually typing errors mean that there is something wrong and needs to be fixed.

Reply:
And I think the error message you pointed, indicate that you did it somehow wrongly.

```
-> Tuple[str, int, str]
```

Is likely what you should have added to the method to have equivalent of :rtype

Reply:
I've double-checked different scenarios and left the `_request` return type as dict because it seems that this is the only case when it works, also .json() method usually returns parsed data in dict type

Reply:
@ephraimbuddy Is it still relevant or we can move forward?)